### PR TITLE
SQL-1999: Update BsonTypeInfo.non_concise_type to be an Option

### PR DESCRIPTION
**Pull request by mattChiaravalloti**
This PR updates the `BsonTypeInfo.non_concise_type` field to be optional.

### CODE REVIEW

File: core/src/bson_type_info.rs [hunk 1]
```diff
@@ -707,7 +710,6 @@ impl BsonTypeInfo {
         if type_mode == TypeMode::Simple && self.simple_type_info.is_some() {
             (self.simple_type_info.clone().unwrap().length)(max_string_length)
         } else {
-            // TODO: is there a reason this was None instead of self.length?
```
Comment (mattChiaravalloti) [inline core/src/bson_type_info.rs:710]: 🚗  drive-by clean-up

**Review by EthanHardyMongo — APPROVED** [review]
